build/pmm-client-docker/entrypoint.py

```diff
@@ -77,13 +78,21 @@ def main():
 
         agent.wait()
 
-        if status != 0:
+        if status != 0 and not PMM_AGENT_SIDECAR:
             sys.exit(status)
 
-    # use execlp to replace the current process (this entrypoint)
-    # with pmm-agent with inherited environment
-    print('Starting pmm-agent ...', file=sys.stderr)
-    os.execlp('pmm-agent', 'run')
+    # restart pmm-agent if PMM_AGENT_SIDECAR flag is provided
+    while True:
+        # use execlp to replace the current process (this entrypoint)
+        # with pmm-agent with inherited environment.
+        print('Starting pmm-agent ...', file=sys.stderr)
+        status = subprocess.call(['pmm-agent', 'run'])
+        print('pmm-agent run exited with {}.'.format(status), file=sys.stderr)
+        if PMM_AGENT_SIDECAR:
+            print('Restarting pmm-agent because PMM_AGENT_SIDECAR is enabled ...', file=sys.stderr)
+            time.sleep(1)
```

Comment:
maybe also do that interval configurable. 1 second probably OK, but who knows maybe we wouldn't be happy with that interval later.

Reply:
ok

Reply:
We think that running in a permanent loop is a performance penalty given that pmm-agent is a 30MB binary and growing.

Reply:
@atymchuk what do you mean? Why do you think that it might be performance penalty?
Permanent loop happens only if PMM_AGENT_SIDECAR is true.

Reply:
Yes, I can see it. Question why do we need to run pmm-agent in a loop? Can we detect it's down and only then launch it?

Reply:
Or maybe you can make it clear here, why this is solved that way?

Reply:
https://docs.python.org/3/library/subprocess.html#subprocess.call

Reply:
How does that link to subprocess answer the question of why we want to run a loop forever?

Reply:
Ok, someone else explained me, sorry for asking questions.

Reply:
Did you read task in Jira? Jira ticket is exactly answer to your question.
First line in the link I shared contains answer to your second question.
> Yes, I can see it. Question is why do we need to run pmm-agent in a loop? Can we detect it's down and only then launch it?
I asked two questions to understand what do you mean and from your message I understand that you think that we constantly trying to start pmm-agent even if it's already run. So I send a link which explains that it's wrong.